src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -941,9 +947,8 @@ internal CSharpTypeInfo GetSpeculativeTypeInfoWorker(int position, ExpressionSyn
                 return CSharpTypeInfo.None;
             }
 
-            Binder binder;
             ImmutableArray<Symbol> crefSymbols;
-            BoundNode boundNode = GetSpeculativelyBoundExpression(position, expression, bindingOption, out binder, out crefSymbols); //calls CheckAndAdjustPosition
+            BoundNode boundNode = GetSpeculativelyBoundExpression(position, expression, bindingOption, out _, out crefSymbols); //calls CheckAndAdjustPosition
```

Comment:
>BoundNode boundNode = GetSpeculativelyBoundExpression(position, expression, bindingOption, out _, out crefSymbols); //calls CheckAndAdjustPosition [](start = 12, length = 146)

It doesn't look like any nullable analysis is going to be done on this node. We should also check other ```GetSpeculative...``` APIs.

Reply:
Speculative semantic models aren't really handled in this pr, other than basic "support" for paths that go through `GetBoundNodes`. That will require more work in a different PR.

---
In reply to: [269810929](https://github.com/dotnet/roslyn/pull/34429#discussion_r269810929) [](ancestors = 269810929)

Reply:
> Speculative semantic models aren't really handled in this pr, other than basic "support" for paths that go through GetBoundNodes. That will require more work in a different PR.

Would it make sense to to not do any analysis for speculative scenarios for now, even in context of member semantic model?

---
In reply to: [271518790](https://github.com/dotnet/roslyn/pull/34429#discussion_r271518790) [](ancestors = 271518790,269810929)

Reply:
I'll add a check in `EnsureBoundRootIfNecessary` to bail out if we're in a speculative model for now.

---
In reply to: [271520509](https://github.com/dotnet/roslyn/pull/34429#discussion_r271520509) [](ancestors = 271520509,271518790,269810929)